r/src/r_to_arrow.cpp

```diff
@@ -727,9 +727,33 @@ class RPrimitiveConverter<T, enable_if_t<is_timestamp_type<T>::value>>
 template <typename T>
 class RPrimitiveConverter<T, enable_if_t<is_decimal_type<T>::value>>
     : public PrimitiveConverter<T, RConverter> {
+  using ValueType = typename arrow::TypeTraits<T>::BuilderType::ValueType;
```

Comment:
```suggestion
  using ValueType = typename arrow::TypeTraits<T>::CType;
```
I think this will work also and is a bit more direct.